cmd/gateway/s3/gateway-s3.go

```diff
@@ -474,6 +482,14 @@ func (l *s3Objects) CopyObject(ctx context.Context, srcBucket string, srcObject
 	// So preserve it by adding "REPLACE" directive to save all the metadata set by CopyObject API.
 	srcInfo.UserDefined["x-amz-metadata-directive"] = "REPLACE"
 	srcInfo.UserDefined["x-amz-copy-source-if-match"] = srcInfo.ETag
+	// See comment above. Tags will be applied on destination object.
+	if !minio.IsStringEqual(srcInfo.UserDefined["X-Amz-Tagging"], "") {
```

Comment:
instead of checking for empty string, you can check for the presence of this key